### PR TITLE
refactor(bbb-web): Use Tika as MIME type detection tool

### PR DESCRIPTION
**Pull request by danielpetri1**
### What does this PR do?
Detects the media type of a given presentation using [Tika](https://tika.apache.org/2.8.0/index.html), which bases its detection method on the document content and potential known file extensions.

The goal is to prevent container formats such as `.docx` and `.pptx` from simply returning e.g. `application/zip` as their `MIME` type, which was occurring when the `file` bash command encountered malformed presentations.

Left as draft so that bringing in Tika as a content analysis toolkit can be discussed with @gustavotrott and @GuiLeme.

### Closes Issue(s)
#17737
#17824
Potentially #17122 (untested)

### More
Removed catch-all antipattern, use of exceptions for regular control flow, and redundant defensive programming. Fixed a few typos throughout related code.

### CODE REVIEW

**Comment by gustavotrott** [comment]
Note for @antobinary,

If we merge this PR, we can remove the dependency of `file` on bbb install.

**Comment by ffdixon** [comment]
There are some sample files in https://github.com/bigbluebutton/bigbluebutton/issues/17737 to test against.

**Review by GuiLeme — CHANGES_REQUESTED** [review]
Just those 2, but overall it's really good!

**Review by GuiLeme — CHANGES_REQUESTED** [review]

File: bbb-common-web/src/main/java/org/bigbluebutton/presentation/SupportedFileTypes.java [hunk 1]
```diff
@@ -24,8 +24,11 @@
 
 import static org.bigbluebutton.presentation.FileTypeConstants.*;
 
+import org.apache.tika.Tika;
 import java.io.BufferedReader;
```
[the remark this answers is not in the document]
Comment (danielpetri1) [reply bbb-common-web/src/main/java/org/bigbluebutton/presentation/SupportedFileTypes.java]: Thanks, fixed.

**Review by GuiLeme — APPROVED** [review]
LGTM!

**Pull request merged**